src/server/aws_lambda.ts

```diff
@@ -49,21 +53,40 @@ function syncHandler<TEvent = any, TResult = any>(handler: SyncHandler<TEvent, T
     return function wrappedLambdaHandler(event, context, cb) {
         hb.__setStore(AsyncStore)
         AsyncStore.run({context: {}, breadcrumbs: []}, () => {
+            const timeoutHandler = setupTimeoutWarning(hb, context)
             try {
                 handler(event, context, (error, result) => {
+                    clearTimeout(timeoutHandler)
                     if (error) {
                         return reportToHoneybadger(hb, error, cb)
                     }
 
                     cb(null, result)
                 });
             } catch (err) {
+                clearTimeout(timeoutHandler)
                 reportToHoneybadger(hb, err, cb)
             }
         })
     }
 }
 
+function shouldReportTimeoutWarning(hb: Honeybadger, context: Context): boolean {
+    return typeof context.getRemainingTimeInMillis === 'function' && !!((hb.config as ServerlessConfig).reportTimeoutWarning)
+}
+
+function setupTimeoutWarning(hb: Honeybadger, context: Context) {
+    if (!shouldReportTimeoutWarning(hb, context)) {
+        return
+    }
+
+    const delay = context.getRemainingTimeInMillis() - ((hb.config as ServerlessConfig).timeoutWarningThresholdMs)
+    return setTimeout(() => {
+        hb.notify(`${context.functionName}[${context.functionVersion}] may timeout`)
```

Comment:
Also, has this been tested in a real function? Asking because I recently encountered sth that worked on a local emulation but failed on AWS. For instance, in a local emulation, the process might hang around for a while, while Lambda might kill it at a point. If you can't test it live, I think their [runtime client](https://github.com/aws/aws-lambda-nodejs-runtime-interface-client) are pretty much how Lambda invokes functions.

Reply:
Yes I [have tested](https://github.com/honeybadger-io/honeybadger-js/pull/758#issuecomment-1100817169)! I also pushed the example lambda function, but if you want to try it you will need to install the build from this branch (until this is merged):
```
cd examples/aws-lambda
npm i honeybadger-io/honeybadger-js#aws_timeout_warning_#679
```

Reply:
Ooh, I missed that comment. All good then.